qiskit/visualization/matplotlib.py

```diff
@@ -608,8 +608,16 @@ def _draw_ops(self, verbose=False):
                                 reg['index'] == carg[1]):
                             c_idxs.append(index)
                             break
-                for ii in q_idxs:
-                    q_anchors[ii].set_index(this_anc, layer_width)
+
+                # if it isn't going to be plotted, don't add it to the anchors. This prevents
+                # additional blank wires at the end of the line if the last instruction
+                # is a barrier type
+                if self.plot_barriers or \
```

Comment:
I'm a bit confused by this shouldn't this be `if not self.plot_barriers`? Right now this doesn't match the comment because we're checking if we're plotting or the instruction is not a barrier type. We want to run this only when it doesn't plot right?

Reply:
The code inside the `if` statement adds the instruction to the list of anchors, and this should only be done if the element is going to be plotted, which will occur if `self.plot_barriers` is true, or the instruction is not a barrier type. I will update the comment to make this clearer, the double negative doesn't help!